rpc/rpc.go

```diff
@@ -65,35 +65,38 @@ func (i *InteropEndpoints) SendTx(signedTx tx.SignedTx) (interface{}, jRPC.Error
 	c.Add(ctx, 1, opts)
 
 	// Check if the RPC is actually registered, if not it won't be possible to assert soundness (in the future once we are stateless won't be needed)
-	if err := i.executor.CheckTx(signedTx); err != nil {
+	if err = i.executor.CheckTx(signedTx); err != nil {
 		return "0x0", jRPC.NewRPCError(jRPC.DefaultErrorCode, fmt.Sprintf("there is no RPC registered for %d", signedTx.Tx.RollupID))
 	}
 
 	// Verify ZKP using eth_call
-	if err := i.executor.Verify(ctx, signedTx); err != nil {
+	if err = i.executor.Verify(ctx, signedTx); err != nil {
 		return "0x0", jRPC.NewRPCError(jRPC.DefaultErrorCode, fmt.Sprintf("failed to verify tx: %s", err))
 	}
 
-	if err := i.executor.Execute(ctx, signedTx); err != nil {
+	if err = i.executor.Execute(ctx, signedTx); err != nil {
 		return "0x0", jRPC.NewRPCError(jRPC.DefaultErrorCode, fmt.Sprintf("failed to execute tx: %s", err))
 	}
 
 	// Send L1 tx
 	dbTx, err := i.db.BeginStateTransaction(ctx)
 	if err != nil {
-		return "0x0", jRPC.NewRPCError(jRPC.DefaultErrorCode, fmt.Sprintf("failed to begin dbTx, error: %s", err))
+		log.Errorf("failed to begin dbTx, error: %s", err)
```

Comment:
every time it returns here, will log an error, so this is redundant?

Reply:
Not necessarily, it returns a short error without details but logs the full error message. It was the main idea of this PR.